packages/gasket-cli/src/scaffold/actions/generate-files.js

```diff
@@ -36,6 +37,34 @@ function reduceDescriptors(descriptors) {
   return Array.from(reduced.values());
 }
 
+/**
+ * Assemble the description objects from glob results
+ *
+ * @param {string} dest - app destination
+ * @param {string} from - Plugin source name
+ * @param {string} pattern - pattern to glob
+ * @param {string[]} srcPaths - resulting paths from glob
+ * @returns {object[]} descriptors
+ */
+function assembleDescriptors(dest, from, pattern, srcPaths) {
+  const output = dest.split(reSep).join(path.sep);
+  const baseParts = path.resolve(pattern.replace(/[/\\]+.?\*.*$/, '')).split(reSep);
+  const base = baseParts.join(path.sep);
+  return srcPaths.map((srcPath) => {
+    const parts = srcPath.split(reSep);
+    const srcFile = parts.join(path.sep);
```

Comment:
Perhaps this split/rejoin could be in its own utility function, like `normalizePathSeparators`?